### PR TITLE
Update to no longer synchronize on Integer objects

### LINKED ISSUE #16299: Fixes issues synchronizing on value based classes
**Describe the bug**
Running Open Liberty with the `-XX:+UnlockDiagnosticVMOptions -XX:DiagnoseSyncOnValueBasedClasses=2` and JDK 16, you'll get a lot of warnings of synchronizing on value based classes, mostly java.lang.Integer

https://openjdk.java.net/jeps/390

An example of the warning:
```
[17,773s][info][valuebasedclasses] Synchronizing on object 0x000000070a000cc0 of klass java.lang.Integer
[17,773s][info][valuebasedclasses]      at com.ibm.ws.artifact.zip.cache.internal.ZipFileReaper.getRipest(ZipFileReaper.java:976)
[17,773s][info][valuebasedclasses]      - locked <0x000000070a000cc0> (a java.lang.Integer)
[17,773s][info][valuebasedclasses]      at com.ibm.ws.artifact.zip.cache.internal.ZipFileReaper$ReaperRunnable.run(ZipFileReaper.java:478)
[17,773s][info][valuebasedclasses]      at java.lang.Thread.run(java.base@16/Thread.java:831)
```

### PR DESCRIPTION
**Pull request by jhanders34**
- With Java 16, JEP 390 changes Integer primitive wrappers so that they should not be synchronized on any longer.

The changes in this PR undo many previous PRs that introduced synchronizing on Integer in order to remove having so many inner classes used for synchronization.  This change re-introduces many of those inner classes again.  This will have a small impact on footprint and probably CPU since we will be loading more classes.

Fixes #16299

### CODE REVIEW

**Comment by jhanders34** [comment]
#build

File: dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/cache/internal/ZipFileHandleImpl.java [hunk 1]
```diff
@@ -88,7 +88,10 @@ public long getLastModified() {
 
     //
 
-    private final Integer zipFileLock = new Integer(0);
+    private class ZipFileLock {
```
Comment (aguibert) [inline dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/cache/internal/ZipFileHandleImpl.java:91]: could this be made a static class to reduce overhead of creating a new `ZipFileHandleImpl` (same comment applies to other instances of occurrences of non-static private lock classes)

File: dev/com.ibm.ws.channelfw/src/com/ibm/ws/bytebuffer/internal/WsByteBufferPoolManagerImpl.java [hunk 2]
```diff
@@ -92,7 +92,7 @@
     private long lastTimeCheck = 0L;
     private int leakDetectionInterval = -1;
     private String leakDetectionOutput = null;
-    private final Integer leakDetectionSyncObject = new Integer(1);
+    private final Object leakDetectionSyncObject = new Object() {};
```
Comment (aguibert) [inline dev/com.ibm.ws.channelfw/src/com/ibm/ws/bytebuffer/internal/WsByteBufferPoolManagerImpl.java:95]: Why switching to `new Object()` here? AFAIK the `new Object()` constructor might be removed in a future Java release, so we should avoid new usages of this.

File: dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/cache/internal/ZipFileHandleImpl.java (hunk 1 above)
Comment (jhanders34) [reply dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/cache/internal/ZipFileHandleImpl.java:91]: I just put it back to what it was before, but sure, it makes sense to make them all static.

File: dev/com.ibm.ws.channelfw/src/com/ibm/ws/bytebuffer/internal/WsByteBufferPoolManagerImpl.java (hunk 2 above)
Comment (jhanders34) [reply dev/com.ibm.ws.channelfw/src/com/ibm/ws/bytebuffer/internal/WsByteBufferPoolManagerImpl.java:95]: It is making an anonymous class that extends Object.  If that won't work, I can make it a static inner class like in the other classes.  I may have some additional classes to change if we want to remove creating anonymous inner classes that way.

**Review by njr-11 — APPROVED** [review]
It's probably just that I don't understand the background regarding the performance aspects of what sort of lock is used, but it seems odd to me that so many different patterns are used when switching away from new instances of Integer.
In some cases, I see a single inner class being defined per lock. In other cases, I see a single inner class being defined for all locks within the main class to use. And in other cases, a new instance of java.lang.Object.  Is one of these patterns better than the others? Can we standardize on something instead of varying between 3 different ways to solve the same problem?  The other question that comes up is why a lock object is needed at all in cases where there is already a final field such as the following,

```
    private static final ZipFileHandlesLock zipFileHandlesLock = new ZipFileHandlesLock();
    private static final LinkedHashMap<String, ZipFileHandle> zipFileHandles;
```

Is it bad performance-wise to synchronize on a LinkedHashMap or other existing instance?
I'll approve this pull because I'm really asking these questions for my own information, and for future code that is written.

**Comment by jhanders34** [comment]
> In some cases, I see a single inner class being defined per lock. In other cases, I see a single inner class being defined for all locks within the main class to use. And in other cases, a new instance of java.lang.Object. Is one of these patterns better than the others? Can we standardize on something instead of varying between 3 different ways to solve the same problem? The other question that comes up is why a lock object is needed at all in cases where there is already a final field such as the following,

In general I tried to put things back to the way that they were.  Previously people had defined an inner class per lock in order to help find the locks when looking at a thread dump.  I used a single one in classes where the type didn't matter.  We could just standardize on a Class in one of the packages or something as well to make it simpler and not having so many inner classes in the same bundle for locking.  Some people don't like that, so I tried to honor how code owners had done it in the past.  I went back to the previous versions of the class where I changed it to Integer and put it back to the previous way.  It sounds like doing Object() {} to make inner classes is frowned upon as well and will move away from that.

> 
> ```
>     private static final ZipFileHandlesLock zipFileHandlesLock = new ZipFileHandlesLock();
>     private static final LinkedHashMap<String, ZipFileHandle> zipFileHandles;
> ```
> 
> Is it bad performance-wise to synchronize on a LinkedHashMap or other existing instance?
> I'll approve this pull because I'm really asking these questions for my own information, and for future code that is written.

That is a valid point.  There doesn't need to be a separate lock for that particular one and I will make a change to not have the separate lock any longer.

**Review by tbitonti — APPROVED** [review]
Looks fine; except the copyright should be "2021".

**Comment by jhanders34** [comment]
#build

**Comment by jhanders34** [comment]
#libby

**Pull request merged**